### PR TITLE
Add checks for Foldable

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

**Comment by garyb** [comment]
I think `Foldable` is technically lawless, but these look like good properties to have even if they're not officially required. Thanks!

**Comment by jacereda** [comment]
I thought it had laws because http://hackage.haskell.org/package/base-4.9.0.0/docs/Data-Foldable.html states that 

> Foldable instances are expected to satisfy the following laws:
> 
> foldr f z t = appEndo (foldMap (Endo . f) t ) z
> foldl f z t = appEndo (getDual (foldMap (Dual . Endo . flip f) t)) z
> fold = foldMap id

and

> If the type is also a Functor instance, it should satisfy
> 
> foldMap f = fold . fmap f

File: src/Test/QuickCheck/Laws/Data/Foldable.purs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  where
+    foldrLaw :: (A -> B -> B) -> B -> f A -> Boolean
+    foldrLaw f z t = foldr f z t == foldrDefault f z t
+
+    foldlLaw :: (B -> A -> B) -> B -> f A -> Boolean
+    foldlLaw f z t = foldl f z t == foldlDefault f z t
+                                                           
+    foldLaw :: f A -> Boolean
+    foldLaw t = fold t == foldMap id t
```
Comment (hdgarrood) [inline src/Test/QuickCheck/Laws/Data/Foldable.purs:39]: This is just the definition of `fold`, though: https://github.com/purescript/purescript-foldable-traversable/blob/17ddb52928dd00fdb31b1cde3931991d6ea9e366/src/Data/Foldable.purs#L166

**Comment by jacereda** [comment]
Oh, right. I'll remove it.

**Pull request merged**